src/vs/workbench/services/workingCopy/common/storedFileWorkingCopy.ts

```diff
@@ -77,7 +77,7 @@ export interface IStoredFileWorkingCopyModel extends IFileWorkingCopyModel {
 	 * which is to ask the model for a `snapshot` and then
 	 * writing that to the model's resource.
 	 */
-	save?(options: IWriteFileOptions, token: CancellationToken): Promise<IFileStatWithMetadata>;
+	getSaveDelegate(): ((options: IWriteFileOptions, token: CancellationToken) => Promise<IFileStatWithMetadata>) | undefined;
```

Comment:
This method should be optional and if defined, should not return `undefined`.

Reply:
Maybe this could be a property rather than a method? And then please also introduce a `IStoredFileWorkingCopySaveDelegate` interface with documentation.

Reply:
There needs to be a way for a defined method to indicate that the delegate should not be used for the fix that I'm trying to implement here. If something requires going async before we collect the notebook data to be saved, we don't want this method to be used.
Or we could rely on the model being resolved (it probably always is) and throw if not.

Reply:
But that can still be a property, which if defined, can indicate this?

Reply:
no more changes to working copy land

Reply:
Nice!